packages/ui/ant-design/src/WalletModal.tsx

```diff
@@ -24,7 +24,14 @@ export const WalletModal: FC<WalletModalProps> = ({
 
     const [featured, more] = useMemo(() => {
         const supportedWallets = wallets.filter((wallet) => wallet.readyState !== WalletReadyState.Unsupported);
-        return [supportedWallets.slice(0, featuredWallets), supportedWallets.slice(featuredWallets)];
+        const installedWallets = supportedWallets.filter((wallet) => wallet.readyState === WalletReadyState.Installed);
+        const loadableWallets = supportedWallets.filter((wallet) => wallet.readyState === WalletReadyState.Loadable);
+        const notDetectedWallets = supportedWallets.filter(
+            (wallet) => wallet.readyState === WalletReadyState.NotDetected
+        );
+        const orderedWallets = [...installedWallets, ...loadableWallets, ...notDetectedWallets];
+
+        return [orderedWallets.slice(0, featuredWallets), orderedWallets.slice(featuredWallets)];
```

Comment:
Do we want to just reuse the logic from the react-ui one where we iterate the array once?

Reply:
Updated, thanks! Also updated the react-ui modal to do the listed/collapsed filtering in the `useMemo`